src/exercises/exerciseTypes/BottomInput.js

```diff
@@ -26,6 +26,7 @@ export default function BottomInput({
   const [isLongerThanSolution, setIsLongerThanSolution] = useState(false);
   const [isInputWrongLanguage, setIsInputWrongLanguage] = useState(false);
   const [feedbackMessage, setFeedbackMessage] = useState("");
+  const [firstHint, setFirstHint] = useState(true);
```

Comment:
I am not sure if ButtonInput should keep track if it's the first hint or not. 

Technically this should be as independent of the parent as possible, and we should handle whatever happens with the hit based on the exercise itself. 

Consider moving this logic to the exercise itself and use the onHintUsed maybe to handle the specific case?